pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -283,45 +283,58 @@ func (s *Reconciler) Update(ctx context.Context) error {
 		Message: machineCreationSucceedMessage,
 	})
 
+	vmState := v1beta1.VMState(*vm.ProvisioningState)
```

Comment:
Original purpose of `isVMExists` was to decide if vm object exists. After the vm object was successfully retrieved, `Exists` went through additional checking. Exists was `true` only when vm was in `VMStateSucceeded` or `VMStateUpdating` state. Though, after `isVMExists` returned true, both s.scope.MachineStatus.VMID and s.scope.MachineStatus.VMState were set. Given that your PR got merged, the workflow of setting `VMID` and `VMState` is different now. `Exists` still returns `false`/`true` the same way. However, `s.scope.MachineStatus.VMID` and `s.scope.MachineStatus.VMState` will not be set until `Update` op is called.`s.scope.MachineStatus.VMState` will always be set to `Succeeded` or `Failed`. It will be empty for the first 40s. The whole purpose of that field was to report in which state during provisioning a vm is. Now it's broken.